contribution-guide.md

```diff
@@ -0,0 +1,143 @@
+    - Use comments on the issue itself to indicate that you will be working on it and get guidance and help.
+
+## Obtaining the source code 
+
+1. Execute the below command to clone the ballerina-lang Git repo.
+```bash 
+git clone --recursive https://github.com/ballerina-platform/ballerina-lang.git
+```
+>**Tip:** If you have already forked the repository to your GitHub account, then execute the below command replacing 'YOUR-GITHUB-USERNAME' with your Git username.
+```bash 
+git clone --recursive https://github.com/<YOUR-GITHUB-USERNAME>/ballerina-lang.git
+```
+2. Execute the below command to update the Git submodules.
+```bash 
+git submodule update --init
+```
+## Building the project
+
+1. Navigate to the <BALLERINA-LANG_REPO> root directory and execute the below Gradle command.
+```bash 
+./gradlew clean build
+```
+2. Extract the built Ballerina distributions in the below locations.
+-  **runtime only:** <BALLERINA-LANG_REPO>/distribution/zip/jballerina/build/distributions/jballerina-<version>-SNAPSHOT.zip. 
+-  **runtime and tools (e.g., Ballerina Composer):** <BALLERINA-LANG_REPO>/distribution/zip/jballerina-tools/build/distributions/jballerina-tools-<version>-SNAPSHOT.zip.
+>**Note:** If you face an IOException error stating "Too many open files", this is due to the default number of possible open files being set to a lower number on your operating system than required for Ballerina to be compiled. You may have to increase the number of open files/file descriptors (FD) on your operating system to 1000000 (or higher).
+
+## Setting up the developer environment
+
+See the below sections to set up IntelliJ IDEA or Eclipse as your development environment.
+
+### Setting up IntelliJ IDEA
+
+Follow the steps below to import the Ballerina project to IntelliJ IDEA similar to any other Gradle project.
+
+1. In IntelliJ, click **File** and then click **Open**.
+2. Browse the file system and select the <BALLERINA-LANG_REPO> root directory.
+>**Tip:** If you are opening the directory for the first time in IntelliJ and if prompted to import it as a Gradle project, select the **Auto import** option.
+3. Set Project SDK as Java 1.8.
+
+### Setting up Eclipse
+
+As all Ballerina repositories are developed as Maven or Gradle projects, follow the steps below to import any Ballerina project to Eclipse similar to any Maven or Gradle project.
+
+1. In Eclipse, click **File** and then click **Import..**.
+2. Under the **Maven** option, select **Existing Maven Projects**. 
+3. Browse the file system and select the Maven module to open it.
+
+## Contributing to Ballerina grammar
+
+Ballerina grammar has been implemented using ANTLR4. To get a basic understanding on ANTLR 4 grammar syntax and concepts before working with Ballerina grammar, go to [Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference](#http://lms.ui.ac.ir/public/group/90/59/01/15738_ce57.pdf).
```

Comment:
Should we be linking to PDFs like this? I'm not sure whether they have the rights to redistribute the PDF like this. Shall we just link it to the publisher's page? https://www.oreilly.com/library/view/the-definitive-antlr/9781941222621/

Reply:
Changed the link.